cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTable.java

```diff
@@ -71,11 +94,49 @@ public HBaseMetricsTable(DatasetContext datasetContext, DatasetSpecification spe
     this.columnFamily = TableProperties.getColumnFamilyBytes(spec.getProperties());
   }
 
+  private void initializeV3Vars(DatasetSpecification spec) {
+    boolean isV3Table = spec.getName().contains("v3");
+    this.scanExecutor = null;
+    this.rowKeyDistributor = null;
+    if (isV3Table) {
+      RejectedExecutionHandler callerRunsPolicy = new RejectedExecutionHandler() {
+
+        @Override
+        public void rejectedExecution(Runnable r, ThreadPoolExecutor executor) {
+          REJECTION_LOG.info(
+            "No more threads in the HBase scan thread pool. Consider increase {}. Scan from caller thread {}",
+            Constants.MessagingSystem.HBASE_MAX_SCAN_THREADS, Thread.currentThread().getName()
```

Comment:
seems like you're overloading a property that has 'messaging' in its name. This can be misleading.
Probably have your own property name, even if the behavior is similar. Or if the common use case is that the value of both would need updating often, rename the property to something that isn't specific to messaging, and then reuse it.